pymc_marketing/clv/models/pareto_nbd.py

```diff
@@ -0,0 +1,589 @@
+                beta,
+                r + x + n,
+                r + x,
+                r + x + 1,
+            )
+        else:
+            min_of_alpha_beta, max_of_alpha_beta, p, _, _ = (
+                beta,
+                alpha,
+                s + 1,
+                s + 1,
+                s,
+            )
+        abs_alpha_beta = max_of_alpha_beta - min_of_alpha_beta
+
+        log_p_zero = (
+            gammaln(r + x)
+            + r * log(alpha)
+            + s * log(beta)
+            - (gammaln(r) + (r + x) * log(alpha + T) + s * log(beta + T) + loglike)
+        )
+        log_B_one = (
+            gammaln(r + x + n)
+            + r * log(alpha)
+            + s * log(beta)
+            - (
+                gammaln(r)
+                + (r + x + n) * log(alpha + T + future_t)
+                + s * log(beta + T + future_t)
+            )
+        )
+        log_B_two = (
+            r * log(alpha)
+            + s * log(beta)
+            + gammaln(r + s + x)
+            + betaln(r + x + n, s + 1)
+            + log(
+                hyp2f1(
+                    r + s + x,
+                    p,
+                    r + s + x + n + 1,
+                    abs_alpha_beta / (max_of_alpha_beta + T),
+                )
+            )
+            - (gammaln(r) + gammaln(s) + (r + s + x) * log(max_of_alpha_beta + T))
+        )
+
+        def _log_B_three(i):
+            return (
+                r * log(alpha)
+                + s * log(beta)
+                + gammaln(r + s + x + i)
+                + betaln(r + x + n, s + 1)
+                + log(
+                    hyp2f1(
+                        r + s + x + i,
+                        p,
+                        r + s + x + n + 1,
+                        abs_alpha_beta / (max_of_alpha_beta + T + future_t),
+                    )
+                )
+                - (
+                    gammaln(r)
+                    + gammaln(s)
+                    + (r + s + x + i) * log(max_of_alpha_beta + T + future_t)
+                )
+            )
+
+        zeroth_term = (n == 0) * (1 - exp(log_p_zero))
+        first_term = n * log(future_t) - gammaln(n + 1) + log_B_one - loglike
+        second_term = log_B_two - loglike
+        third_term = logsumexp(
+            [
+                i * log(future_t) - gammaln(i + 1) + _log_B_three(i) - loglike
+                for i in range(n + 1)
+            ],
+            axis=0,
+        )
+
+        try:
+            size = len(x)
+            sign = np.ones(size)
+        except TypeError:
+            sign = 1
+
+        # In some scenarios (e.g. large n) tiny numerical errors in the calculation of second_term and third_term
+        # cause sumexp to be ever so slightly negative and logsumexp throws an error. Hence we ignore the sign here.
+        purchase_prob = zeroth_term + exp(
+            logsumexp(
+                [first_term, second_term, third_term],
+                b=[sign, sign, -sign],
```

Comment:
The problem here is that `[first_term, second_term, third_term]` have shape `(3, ...)`, and `b` has shape `(3,)`. Broadcasting works from the right not the left. 

You can either concatenate the terms along the last axis, so they have shape `(..., 3)` or add dummy dimensions to `b` so that it has shape `(3, 1, ..., 1)`.

Also if you only care about the result, you could set `return_sign=False` and you won't have to index `[0]`